src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Nuget/ProjectRestoreInfoBuilder.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using NuGet.SolutionRestoreManager;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Nuget
+{
+    internal static class ProjectRestoreInfoBuilder
+    {
+        internal static IVsProjectRestoreInfo Build(ImmutableList<IProjectValueVersions> updates)
+        {
+            string baseIntermediatePath = null;
+            var targetFrameworks = new TargetFrameworks();
+            
+            foreach (IProjectVersionedValue<IProjectSubscriptionUpdate> update in updates)
+            {
+                var configurationChanges = update.Value.ProjectChanges[ConfigurationGeneral.SchemaName];
+                baseIntermediatePath = baseIntermediatePath ?? configurationChanges.After.Properties["BaseIntermediateOutputPath"];
+                string targetFrameworkMoniker = configurationChanges.After.Properties["TargetFrameworkMoniker"];
```

Comment:
Please use the constants from the Rules themselves representing TargetFrameworkMoniker and BaseIntermediateOutputPath.